### PR TITLE
Add interface for transaction state query

### PR DESCRIPTION
**Pull request by leventeliu**

### CODE REVIEW

File: blockproducer/rpc.go [hunk 1]
```diff
@@ -139,6 +134,18 @@ func (s *ChainRPCService) QuerySQLChainProfile(req *types.QuerySQLChainProfileRe
 	return
 }
 
+func (s *ChainRPCService) QueryTxState(
```
Comment (covenantdb) [inline blockproducer/rpc.go:129]: **[golint]** <sub>reported by [reviewdog](https://github.com/haya14busa/reviewdog) :dog:</sub>
exported method ChainRPCService.QueryTxState should have comment or be unexported

File: types/bprpc.go [hunk 2]
```diff
@@ -181,3 +185,14 @@ type QuerySQLChainProfileResp struct {
 	proto.Envelope
 	Profile SQLChainProfile
 }
+
+type QueryTxStateReq struct {
```
Comment (covenantdb) [inline types/bprpc.go:177]: **[golint]** <sub>reported by [reviewdog](https://github.com/haya14busa/reviewdog) :dog:</sub>
exported type QueryTxStateReq should have comment or be unexported

File: types/bprpc.go [hunk 3]
```diff
@@ -181,3 +185,14 @@ type QuerySQLChainProfileResp struct {
...
+	proto.Envelope
+	Hash hash.Hash
+}
+
+type QueryTxStateResp struct {
```
Comment (covenantdb) [inline types/bprpc.go:183]: **[golint]** <sub>reported by [reviewdog](https://github.com/haya14busa/reviewdog) :dog:</sub>
exported type QueryTxStateResp should have comment or be unexported

File: blockproducer/bpinfo.go [hunk 4]
```diff
@@ -0,0 +1,68 @@
...
+	role   string
+	nodeID proto.NodeID
+}
+
+// String implements fmt.Stringer.
+func (i *blockProducerInfo) String() string {
+	return fmt.Sprintf("[%d/%d|%s] %s", i.rank+1, i.total, i.role, i.nodeID)
+}
+
+func newBlockProduerInfos(
```
Comment (ggicci) [inline blockproducer/bpinfo.go]: Please rename it to `buildBlockProduerInfos`

**Review by zeqing-guo — APPROVED** [review]
Approve

File: blockproducer/branch.go [hunk 5]
```diff
@@ -58,7 +57,7 @@ func newBranch(
 	}
 	// Apply new blocks to view and pool
 	for _, bn := range list {
-		if len(bn.block.Transactions) > transactionsLimit {
+		if len(bn.block.Transactions) > pl.MaxPendingTxsPerAccount {
```
Comment (xq262144) [inline blockproducer/branch.go]: Typo here ?

File: blockproducer/metastate.go [hunk 6]
```diff
@@ -106,7 +106,7 @@ func (s *metaState) loadAccountTokenBalance(addr proto.AccountAddress,
 		log.WithFields(log.Fields{
 			"account":   addr.String(),
 			"balance":   b,
-			"tokenType": tokenType.String(),
+			"tokenType": tokenType,
```
Comment (xq262144) [inline blockproducer/metastate.go]: Unified as `token_type` or `tokenType`

**Review by xq262144 — APPROVED** [review]
Approve

**Pull request merged**